engine/src/main/java/com/datatorrent/stram/StramLocalCluster.java

```diff
@@ -283,21 +282,32 @@ public StramLocalCluster(LogicalPlan dag) throws IOException, ClassNotFoundExcep
     dag.validate();
     // ensure plan can be serialized
     cloneLogicalPlan(dag);
-    // convert to URI so we always write to local file system,
-    // even when the environment has a default HDFS location.
-    String pathUri = CLUSTER_WORK_DIR.toURI().toString();
+    final Path pathUri;
+    String appPath = dag.getAttributes().get(LogicalPlan.APPLICATION_PATH);
+    if (appPath == null) {
+      // convert to URI so we always write to local file system,
+      // even when the environment has a default HDFS location.
+      // assumes execution as unit test
+      pathUri = new Path(DEFAULT_APP_DIR.toURI());
+      dag.getAttributes().put(LogicalPlan.APPLICATION_PATH, pathUri.toString());
+    } else {
+      // should accept any valid path URI (or relative path) provided by user
+      Path tmp = new Path(appPath);
+      if (!tmp.isAbsolute()) {
+        pathUri = new Path(new File(appPath).toURI());
```

Comment:
Is it necessary to update dag `LogicalPlan.APPLICATION_PATH` attribute with URI path?